tests/lib/nested.sh

```diff
@@ -232,9 +232,32 @@ create_nested_core_vm(){
 
             snap download --basename=pc-kernel --channel="20/edge" pc-kernel
             uc20_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$WORK_DIR/image"
-            EXTRA_FUNDAMENTAL="--snap $WORK_DIR/image/pc-kernel_*.snap"
-            chmod 0600 "$WORK_DIR"/image/pc-kernel_*.snap
+
+            # Prepare the pc kernel snap
+            KERNEL_SNAP=$(ls "$WORK_DIR"/image/pc-kernel_*.snap)
+            KERNEL_UNPACKED="$WORK_DIR"/image/kernel-unpacked
+            unsquashfs -d "$KERNEL_UNPACKED" "$KERNEL_SNAP"
+            sbattach --remove "$KERNEL_UNPACKED/kernel.efi"
+            sbsign --key /etc/ssl/private/ssl-cert-snakeoil.key --cert /etc/ssl/certs/ssl-cert-snakeoil.pem "$KERNEL_UNPACKED/kernel.efi"  --output "$KERNEL_UNPACKED/kernel.efi"
```

Comment:
shouldn't we use the same keys as for the shim?

Reply:
@anonymouse64 is it possible? is should be easier to do that. I'll make a try now.

Reply:
Seems to work but now I see this error:
user1@ubuntu:~$ test-snapd-mokutil --sb-state
snap-confine has elevated permissions and is not confined but should be. Refusing to continue to avoid permission escalation attacks: Operation not permitted

Not sure if it could be related, but it could be for a change in snapd as well.

Reply:
tbh I'm not sure why these keys aren't the same, I thought that all the snakeoil keys were the same but they have different hashes and I didn't spend more time looking into why the two keys aren't the same, perhaps they just have a different formatting but the key content is actually the same?

Reply:
@sergiocazzolato re the snap-confine error no that's an unrelated error, if the keys didn't match you wouldn't be able to boot at all, so the fact that you can boot means that the keys do in fact match. I don't know why snap-confine would have elevated permissions, did you manually re-pack the snapd snap for this run? If so perhaps `snap pack` messed up the permissions of the snapd snap?